actionmailer/lib/action_mailer/base.rb

```diff
@@ -900,15 +913,19 @@ def collect_responses(headers)
           yield(collector)
           collector.responses
         elsif headers[:body]
-          [{
-            body: headers.delete(:body),
-            content_type: self.class.default[:content_type] || "text/plain"
-          }]
+          collect_responses_from_text(headers)
         else
           collect_responses_from_templates(headers)
         end
       end
 
+      def collect_responses_from_text(headers)
+        [{
+          body: headers.delete(:body),
+          content_type: headers[:content_type] || self.class.default[:content_type] || "text/plain"
```

Comment:
This method is only called after the `apply_defaults` method is called and the `headers` passed to this method as argument is the result of `apply_defaults`, so unless `apply_defaults` is broken. `self.class.default[:content_type]` should be already applied on `headers[:content_type]`.